server/services/party/party.service.ts

```diff
@@ -48,11 +51,23 @@ export class PartyService {
         }
     }
 
-    async pickLeader(data: PartyLeaderOnline, party: Party): Promise<string> {
+    async makeLeader(leaderId:number, characterId:number) {
+        const party = await this.repo.findOne({leader: leaderId})
+        const character = await this.character.getCharacter(characterId)
+        if (party.members.includes(characterId) && character.status === 'online') {
```

Comment:
We will shoot ourselves in the foot if we allow users to set leaders to offline players.